src/Type/TypeCombinator.php

```diff
@@ -391,7 +391,7 @@ public static function union(Type ...$types): Type
 				unset($tempTypes[$i]);
 			}
 
-			if (count($tempTypes) === 0) {
```

Comment:
There is no significant performance gain between them.

See: https://github.com/symfony/symfony/pull/43779

Reply:
Yes, I also tested it and had a quick research and `count()` is O(1) since the number of elements is stored in the
HashTable. So, the performance benefit in this PR was maybe from not interacting with the array at all?

Reply:
Exactly